src/plugins/telemetry/server/telemetry_collection/ingest_solutions/ingest_solutions.ts

```diff
@@ -0,0 +1,175 @@
+    // Filter out the system indices unless they are required by the patterns
+    if (
+      indexName.startsWith('.') &&
+      !startingDotPatternsUntilTheFirstAsterisk.find(pattern => indexName.startsWith(pattern))
+    ) {
+      return acc;
+    }
+    const matchingTechnology = INGEST_SOLUTIONS.find(({ pattern }) => {
+      if (!pattern.startsWith('.') && indexName.startsWith('.')) {
+        // avoid system indices caught by very fuzzy index patters (i.e.: *log* would catch `.kibana-log-...`)
+        return false;
+      }
+      return new RegExp(`^${pattern.replace(/\./g, '\\.').replace(/\*/g, '.*')}$`).test(indexName);
+    });
+
+    if (!matchingTechnology) {
+      return acc;
+    }
+    const { name } = matchingTechnology;
+    const dataProviders = acc.data_providers || {};
+    return {
+      ...acc,
+      data_providers: {
+        ...dataProviders,
+        [name]: {
+          ...dataProviders[name],
+          index_count: (dataProviders[name]?.index_count || 0) + 1,
+          ...(typeof isECS === 'boolean'
+            ? {
+                ecs_index_count: (dataProviders[name]?.ecs_index_count || 0) + (isECS ? 1 : 0),
+              }
+            : {}),
+          ...(docCount ? { doc_count: (dataProviders[name]?.doc_count || 0) + docCount } : {}),
+          ...(sizeInBytes
+            ? { size_in_bytes: (dataProviders[name]?.size_in_bytes || 0) + sizeInBytes }
+            : {}),
+        },
+      },
+    };
+  }, {} as IngestSolutionsPayload);
+}
+
+interface IndexStats {
+  indices: {
+    [indexName: string]: {
+      total: {
+        docs: {
+          count: number;
+          deleted: number;
+        };
+        store: {
+          size_in_bytes: number;
+        };
+      };
+    };
+  };
+}
+
+interface ClusterState {
+  metadata: {
+    indices: {
+      [indexName: string]: {
+        version: number;
+        mappings: {
+          _doc?: {
+            properties: {
+              ecs?: {
+                properties: {
+                  version?: {
+                    type: string;
+                  };
+                };
+              };
+            };
+          };
+          [_type: string]: object | undefined;
+        };
+      };
+    };
+  };
+}
+
+export async function getIngestSolutions(callCluster: APICaller) {
+  const index = INGEST_SOLUTIONS.map(({ pattern }) => pattern);
+  const [state, indexStats]: [ClusterState, IndexStats] = await Promise.all([
+    // GET _cluster/state/metadata/<index>?filter_path=metadata.indices.*.version
+    callCluster<ClusterState>('cluster.state', {
+      index,
+      metric: 'metadata',
+      filterPath: [
+        // The payload is huge and we are only after the name (no other useful stuff so far)
+        'metadata.indices.*.version',
+        // Does it have `ecs.version` in the mappings?
+        'metadata.indices.*.mappings._doc.properties.ecs.properties.version.type',
+      ],
+    }),
```

Comment:
There is a concern about using this API because [its documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-state.html) says:
> The response is an internal representation of the cluster state and its format may change from version to version. If possible, you should obtain any information from the cluster state using the other, more stable, [cluster APIs](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster.html).

I've added a functional test in `test/api_integration/apis/telemetry/telemetry_local.js` to make sure it works as expected. Although it brings up the risk of flaky tests in the future if the API changes the way it works.

Reply:
How much tech debt do we want to get ourselves into? If it is possible to use a more stable API, I think it's worth changing.

Reply:
I think the only alternative so far is to only use the `GET <index>/_stats/` API. But that requires the `kibana` user to have a more permissive role.

Alternatively, there are some ongoing talks with the ES team to modify the same Cluster State API to provide the aggregated data all at once (meaning ES shouldn't drop support or make any changes in that API).

Reply:
Can we discuss with other teams the possiblity of adding those permissions to the kibana role instead of using this API? im not completely against it but i think it would make more sense to add those extra permissions.


Also if you check the compatiblity grid: https://github.com/elastic/kibana/#version-compatibility-with-elasticsearch

Are we sure this API does not change behavior across the compatiblity grid? Our tests will test exact ES matching version, but not other compatible versions where ES minor/patch versinos are newer or ES patch version is lower than kibana's.

Reply:
I'm not against updating the roles to be more permissive (that will allow us to consistently be able to retrieve the `doc_count` and `size_in_bytes` properties) but it involves some additional security concerns.

Not being able to retrieve the data from the cluster state API because of the compatibility grid will only result in not being able to provide this parameter in the telemetry (it will return it as `{}`) but it shouldn't break any other logic unless the API request itself throws any errors. 
The warning in that API is about the format may change though, not the API behaviour as such.

N.B.: I just pushed a commit to catch the method and safely return `{}` if any of the API calls fail.

I think this approach is safer than opening the `kibana_system` role to be able to read from any index. But I'm happy to revisit this implementation if we think that's the way to go or any other approach (like ES providing this kind of info already embedded in the `_cluster/stats` API if they are willing the do the change and aggregation on their end).

Reply:
Opening up the `kibana_system` role is not ideal. Could we explore creating a `telemetry_system` role that has all the permissions needed?

Reply:
I don't think creating a `telemetry_system` roles would necessarily help us here... If we added the `telemetry_system` role to the `kibana_system` user we'd have an equivalent "threat profile". If we created a new `telemetry_system` user which had the `telemetry_system` role, it would make setup more complicated and have a slightly different "threat profile" as both user's credentials would be stored in the `kibana.yml`.

Augmenting the `_cluster/stats` API so we don't have to change the `kibana_system` privileges at all would be the safest option.

However, if we had to give the `kibana_system` role the `monitor` privilege on indices so we can use the index stats API, it's a lot safer than giving it access to read the documents themselves.